dp-cluster-setup-utility.py

```diff
@@ -908,20 +930,48 @@ def __init__(self, base_url, knox_user, knox_group, topology_directory='/etc/kno
     self.knox_group = knox_group
     self.topology_directory = self._check_dir(topology_directory)
 
-  def _check_dir(self, topology_directory):
-    if not os.path.isdir(topology_directory):
-      raise RuntimeError('Knox topology directory does not exist: %s' % topology_directory)
-    return topology_directory
+  def _check_dir(self, knox_artifact_directory, artifact_type='topology'):
+    if not os.path.isdir(knox_artifact_directory):
+      raise RuntimeError('Knox %s directory does not exist: %s' % (artifact_type, knox_artifact_directory))
+    return knox_artifact_directory
+
+  def _check_file(self, service_file):
+    if not os.path.isfile(service_file):
+      raise RuntimeError('Knox service file does not exist: %s' % service_file)
+    return service_file
+
+  def _chown_to_knox(self, path_name):
+    os.chown(path_name, pwd.getpwnam(self.knox_user).pw_uid, grp.getgrnam(self.knox_group).gr_gid)
 
   def add_topology(self, topology_name, content):
     target = '%s/%s.xml' % (self.topology_directory, topology_name)
     print 'Saving topology %s' % target
     with open(target, 'w') as f: f.write(content)
     print '  Changing ownership of %s to %s:%s.' % (topology_name, self.knox_user, self.knox_group)
-    os.chown(target, pwd.getpwnam(self.knox_user).pw_uid, grp.getgrnam(self.knox_group).gr_gid)
+    self._chown_to_knox(target)
     print '  Changing permissions of %s to %o.' % (topology_name, 0644)
     os.chmod(target, 0644)
 
+  def _create_service_file(self, service_dir, file_name):
+    dest_file = '%s/%s' % (service_dir, file_name)
+    src_file = self._check_file('%s/services/profiler-agent/1.0.0/%s' % (os.path.dirname(os.path.realpath(__file__)), file_name))
+    copyfile(src_file, dest_file)
+    self._chown_to_knox(dest_file)
+
+  def update_profiler_agent_service_def(self, current_stack_version):
+    dest_services_base_dir = self._check_dir('/var/lib/knox/data-%s/services' % current_stack_version, 'service')
+    service_dir = '%s/profiler-agent/1.0.0' % dest_services_base_dir
+    if os.path.isdir(service_dir):
+      print 'Service files already exist in %s' % service_dir
+    if not os.path.isdir(service_dir):
```

Comment:
maybe just using _else:_ instead of checking the directory again would be simpler